### PR TITLE
Support for password protected registries

### PR DESCRIPTION
**Pull request by kekru**
Hi Halil,

I added parameters to specify a username and password when the registry is password protected (basic auth). Hope you like it :)

And I added a basic test-setup with a few small test cases for cleaning a registry with and without password protection.

The tests can be run on Travis CI, but maybe we should switch to another service, if they don't behave stable.
Locally the tests are running stable, but on Travis I think the Ubuntu 14.04 isn't the best choice for an up-to-date docker installation ;)

Best regards
Kevin

### CODE REVIEW

**Comment by hcguersoy** [comment]
Hi Kevin,

thank you for your contribution. I'll take a look on it (hopefully this week) and check it.

File: Dockerfile [hunk 1]
```diff
@@ -2,5 +2,6 @@ FROM python:2.7-alpine
 RUN pip install requests
 COPY cleanreg.py /cleanreg.py
 COPY LICENSE /LICENSE
+RUN /cleanreg.py -h
```
Comment (hcguersoy) [inline Dockerfile]: Why did you call here cleanreg?

File: README.md [hunk 2]
```diff
@@ -63,6 +64,13 @@ optional arguments:
                         Ignore tag, or more correct, a digest, if it is
                         referenced multiple times. ATTENTION: the default if
                         False!
+  -u BASICAUTHUSER, --basicauth-user BASICAUTHUSER
+                        The username, if the registry is protected with basic
+                        auth
+  -pw BASICAUTHPW, --basicauth-pw BASICAUTHPW
```
Comment (hcguersoy) [inline README.md:70]: Maybe we should make it possible to provide the user and password via environment variables and docker secrets if running in a container

Comment (hcguersoy) [reply README.md:70]: can you add an example in the readme, too?
And a small part how to run the tests locally.
And please add yourself to the authors part in the py file.

File: Dockerfile (hunk 1 above)
Comment (kekru) [reply Dockerfile]: I found it useful to have a simple syntax error check on build-time, by running the script. But yeah, it's bad style to have this empty layer in the app's Dockerimage. I reverted the commit.

File: README.md (hunk 2 above)
Comment (kekru) [reply README.md:70]: Ok done :)

**Pull request merged**